src/types/utils.ts

```diff
@@ -1,5 +1,13 @@
-import { DirectoryData } from "./directory"
+import { DirectoryData, MediaData } from "./directory"
 
 export const isDirData = (data: unknown): data is DirectoryData => {
   return (data as DirectoryData).type === "dir"
 }
+
+// NOTE: This assumes that mediaPath is proof that this is a media file.
+// The corresponding backend code (mediaFiles.js) does not appear to guarantee the existence of
+// this property on media files (or indeed, ANY property) and this is a best effort attempt,
+// which might fail in the future if new media files are discovered to be returned w/o mediaPath.
+export const isMediaData = (data: unknown): data is MediaData => {
+  return (data as MediaData).type === "file" && !!(data as MediaData).mediaPath
+}
```

Comment:
Just checking, this is only for file data and not media directory data right?

Reply:
Hmm do you happen to have an example where the backed code returns a mediaPath without having a media file? While it would be beyond the scope of this PR - I feel that we can minimally document this bug as an issue as this front-end fix seems to me like a temporary workaround

Reply:
Hmm do you happen to have an example where the backed code returns a mediaPath without having a media file? While it would be beyond the scope of this PR - I feel that we can minimally document this bug as an issue as this front-end fix seems to me like a temporary workaround

Reply:
@alexanderleegs - yep! i did a `filter` to ensure that the types are correct

@kishore03109 - actually i **don't**. it would be a bug if we expect the property to exist but it doesn't. for this case, my confusion stems from `MediaFileService.js`, where we do a `find` operation across an array of objects. this doesn't guarantee existence of the object (leading to `sha` to be `undefined`). 

the BE does guarantee that `mediaPath` exists but this might not be true going forward and hence, my comment is more of a warning than proof of a bug ._.